### PR TITLE
Enforce Strict Code Conventions

### PR DESCRIPTION
**Pull request by Lekesoldat**
**tl;dr**:
I believe it could be beneficial to have stricter code conventions automatically enforced 😊 As of now, there are numerous different coding styles, linting- and formatting rules used, causing major diff's in PR's, commits, and so on.

**Identified problems**:
- Many developers with various coding styles and experience work on the project.
- Some use code-beautifiers, others don't. Results in massive pull requests with thousands of changes because of automated formatting.
- There are different rules for the `frontend` and `backend`. Have not identified a reason for this.

**Solution**:
- [x] Add `pre-commit` hooks not allowing commits of files breaking with specified rules and conventions.
- [x] Create corresponding linting and formatting rules for `frontend` and `backend`.
- [x] Add suggested extensions and settings for easier development according to rules.

### CODE REVIEW

**Comment by Lekesoldat** [comment]
All tests will pass once #220 is merged.
Separated into two pull requests to split the configuration from the formatting

**Comment by Lekesoldat** [comment]
Welp, seems like all files appeared in the pr regardless.
Those of interest are:

**Dokumentasjon**
- `README.md`

**Workflows**
- `.github/workflows/frontend.yml`

**Frontend settings**
- `.husky/pre-commit`
- `.eslintrc.js`
- `.prettierrc`
- `.vscode/extensions.json`
- `.vscode/settings.json`

**Backend settings**
- `.github/workflows/backend.yml`
- ``
- - ``
- - ``

**Comment by AudunBerget** [comment]
Har sett over filene av interesse. Ser greit ut. Har testet lokalt, og det funket - veldig kult. Har ikke testet VSCode, men det kan du helt sikkert gjøre selv. 

Ellers må merge conflicten løses nå :)

**Comment by Lekesoldat** [comment]
> Har sett over filene av interesse. Ser greit ut. Har testet lokalt, og det funket - veldig kult. Har ikke testet VSCode, men det kan du helt sikkert gjøre selv.
> 
> Ellers må merge conflicten løses nå :)

Den er god, tar en titt på det asap!

**Review by AudunBerget — APPROVED** [review]
